### PR TITLE
Add warnings to one gcc compiler in Travis configuration

### PR DESCRIPTION
**Pull request by davidbeckingsale**
This will fail if code is generating warnings.

### CODE REVIEW

**Comment by rhornung67** [comment]
:+1: Thanks!

**Comment by DavidPoliakoff** [comment]
👍  neet

**Comment by rhornung67** [comment]
Wait. How does the Travis log file help me? I don't see any warning/errors, only that the compiler 'cannot build a simple test program'.

**Comment by DavidPoliakoff** [comment]
Odds are he's still wrestling with CMake nastiness

**Comment by davidbeckingsale** [comment]
Correct! There's an error in my Travis config. If I build locally I see 
things like:

     error: comparison between signed and unsigned integer expressions 
[-Werror=sign-compare]

**Comment by rhornung67** [comment]
Sorry. I'm used to thinking that pull requests are ready to review. But that doesn't make sense in  this case.

**Comment by davidbeckingsale** [comment]
No problem, I should have made a note of that. Travis needs the pull request to trigger the build, which is why I created it.

**Comment by davidbeckingsale** [comment]
I'm going to commit both the new .travis.yml and fixes for all the warnings before merging, otherwise our travis builds for develop will start failing.

**Comment by DavidPoliakoff** [comment]
In which Beckingsale gets tricked into fixing everything wrong in raja

**Comment by davidbeckingsale** [comment]
Can you spot what's wrong with my .travis.yml - I'm obviously missing something obvious :(

**Comment by DavidPoliakoff** [comment]
Looks like extra quotes on cmake line. More when not on bus

**Comment by DavidPoliakoff** [comment]
Be sure to remove the echo before commit. Small thing, but would recommend

**Comment by davidbeckingsale** [comment]
@trws do you have a second to take a look at my many attempts to pass extra flags to CMake? 😄  I based it off your coverage branch, but it seems like quoting is going awry and CMake actually thinks the flags are part of the compiler command.

**Comment by trws** [comment]
What you have in the last one looks like it should work, if that doesn't 
you may just have to break down and use `export` in the before_install 
section to get it in there, or set CMAKE_CXX_FLAGS_ENV and have that on 
the cmake line or similar.  Seems like something that should just work, 
but looks like it's a persistent problem for travis users.

**Comment by davidbeckingsale** [comment]
I've tweaked the CMake config to enable warnings as errors when `-DRAJA_ENABLE_WARNINGS=On` is passed to CMake, and enabled this for all CMake builds. There are now plenty of warnings (errors) throughout RAJA 😄

**Comment by rhornung67** [comment]
@davidbeckingsale I just pushed a bunch of commits so that the code (and proxy apps) compiles cleanly for the most. There still a bit more work to do when -Wextra is also used. I'll try to finish that soon. I'm pursuing work therapy in an attempt to forget about the election results...

**Comment by davidbeckingsale** [comment]
This is looking good - thanks for adding the extra host-config too.

File: include/RAJA/internal/defines.hxx [hunk 1]
```diff
@@ -24,19 +24,66 @@
...
+ *  double myVar = ...
+ *  RAJA_DEBUG_VAR(myVar); 
+ *  
+ *                       
+ *  cassert(myVar > 0)  // variable used in an assertion
+ *
+ * \endcode
+ *******************************************************************************
+ */
+#define ATK_DEBUG_VAR(_x)   static_cast<void>(_x)
```
Comment (davidbeckingsale) [inline include/RAJA/internal/defines.hxx]: This needs renaming 😄

Comment (rhornung67) [inline include/RAJA/internal/defines.hxx]: Yeah, I know. I'm working on more changes....

**Pull request merged**